### PR TITLE
DOC: Optimize: update DE documentation

### LINKED ISSUE #17398: BUG: Documentation for `scipy.optimize.differential_evolution` is incomplete, reference link is broken
### Describe your issue.

The different possible choices for [`scipy.optimize.differential_evolution`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.differential_evolution.html), parameter `strategy`, are not explained in the documentation. Currently, it lists 12 different values:

```
    strategy : str, optional
        The differential evolution strategy to use. Should be one of:
            - 'best1bin'
            - 'best1exp'
            - 'rand1exp'
            - 'randtobest1exp'
            - 'currenttobest1exp'
            - 'best2exp'
            - 'rand2exp'
            - 'randtobest1bin'
            - 'currenttobest1bin'
            - 'best2bin'
            - 'rand2bin'
            - 'rand1bin'
        The default is 'best1bin'.
```

Only one of them, the default, is explained in the doc string section **Notes**. There is [reference `[2]`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.differential_evolution.html#r108fc14fa019-2) but the provided link is broken. Also [reference `[1]`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.differential_evolution.html#r108fc14fa019-1) does not seem to contain details on what the different strategies mean.
Since this is a critical parameter of the function, it would be good if it is documented properly, so that users can make an informed choice.

### Reproducing Code Example

```python
This is a docs issue.
```

### Error message

```shell
This is a docs issue.
```

### SciPy/NumPy/Python version information

v1.9.3

### PR DESCRIPTION
**Pull request by mkiffer**
#### Reference issue
Closes #17398

#### Additional information

Documentation has been added regarding the use of evolution strategies. 

I believe this update may need some assistance.  I was not able to check my documentation fits the style requirements. I will submit this pull request and welcome any and all feedback. 
I used the paper as posted by @andyfaff which had some fantastic information. However I am not totally clear on the differences between binomial and exponential implementation of DE optimisation.

### CODE REVIEW

File: scipy/optimize/_differentialevolution.py [hunk 1]
```diff
@@ -294,6 +295,53 @@ def differential_evolution(func, bounds, args=(), strategy='best1bin',
...
+
+            DE/current-to-best/1 : v_i = x_i + F_{cr}(x_b − x_i)
+                                    + F_{xc}(x_{r_1} − x_{r_2})
+
+            DE/rand-to-best/1 : v_i = x_{r_1} + F_{cr}(x_b − x_i)
+                                    + F_{xc}(x_{r_2} − x_{r_3})
+
+
+        where the integers :math:`r1, r2, r3, r4 and r5` are chosen randomly 
+        from the interval [1, NP] and are different from the
```
Comment (andyfaff) [inline scipy/optimize/_differentialevolution.py:320]: `NP` is defined a couple lines below. I think it's worth defining what NP is around about L80.

File: scipy/optimize/_differentialevolution.py [hunk 2]
```diff
@@ -294,6 +295,53 @@ def differential_evolution(func, bounds, args=(), strategy='best1bin',
     iteration, rather than multiple times for all the population members; the
     interpreter overhead is reduced.
 
+    An outline of the strategies and their stengths and weaknesses is given by 
+    Qiang and Mitchell (2014) [8]_
+
+    ..math::
+            DE/rand/1 : v_i = x_{r_1} + F_{xc}(x_{r_2} − x_{r_3})
```
Comment (andyfaff) [inline scipy/optimize/_differentialevolution.py:302]: In equation on line ~260 we use a different notation. Can you convert all the equations you've added to use the same notation?
The reason is that $F_{xc}$ is actually the `mutation` parameter, $F_{cr}$ is the `recombination` parameter, etc.
 I'd like to use terminology that is actually used in the parameters.

File: scipy/optimize/_differentialevolution.py [hunk 3]
```diff
@@ -294,6 +295,53 @@ def differential_evolution(func, bounds, args=(), strategy='best1bin',
...
+
+            DE/rand/2 : v_i = x_{r_1} + F_{xc}(x_{r_2} − x_{r_3}) 
+                                    + F_{xc}(x_{r_4} − x_{r_5})
+
+            DE/best/1 : v_i = x_b + F_{xc}(x_{r_1} − x_{r_2})
+
+            DE/best/2 : v_i = x_b + F_{xc}(x_{r_1} − x_{r_2})
+                                    + F_{xc}(x_{r_3 − x_{r_4})
+
+            DE/current-to-best/1 : v_i = x_i + F_{cr}(x_b − x_i)
```
Comment (andyfaff) [inline scipy/optimize/_differentialevolution.py:312]: The `strategy` str is `'currenttobest*'`.  i.e. the `DE/current-to-best/1` terminology isn't a 1:1 correspondence with a value you can actually use for `strategy`. Should we use `DE/currenttobest/1` instead?

The same goes for all equations here.

File: scipy/optimize/_differentialevolution.py [hunk 4]
```diff
@@ -294,6 +295,53 @@ def differential_evolution(func, bounds, args=(), strategy='best1bin',
     iteration, rather than multiple times for all the population members; the
     interpreter overhead is reduced.
 
+    An outline of the strategies and their stengths and weaknesses is given by
```
Comment (andyfaff) [inline scipy/optimize/_differentialevolution.py:298]: I'd like for the text you've added to come a bit higher up, around L276. It makes more sense there.

Comment (andyfaff) [inline scipy/optimize/_differentialevolution.py:298]: `An outline of the other strategies and their strengths and weaknesses is given by ...`

Comment (andyfaff) [inline scipy/optimize/_differentialevolution.py:298]: stength --> strength

File: scipy/optimize/_differentialevolution.py [hunk 5]
```diff
@@ -294,6 +295,53 @@ def differential_evolution(func, bounds, args=(), strategy='best1bin',
     iteration, rather than multiple times for all the population members; the
     interpreter overhead is reduced.
 
+    An outline of the strategies and their stengths and weaknesses is given by 
+    Qiang and Mitchell (2014) [8]_
```
Comment (andyfaff) [inline scipy/optimize/_differentialevolution.py:299]: We should replace reference 3 on L252 (and elsewhere) with this reference, can you do that?

File: scipy/optimize/_differentialevolution.py [hunk 6]
```diff
@@ -294,6 +295,53 @@ def differential_evolution(func, bounds, args=(), strategy='best1bin',
...
+
+    ..math::
+            DE/rand/1 : v_i = x_{r_1} + F_{xc}(x_{r_2} − x_{r_3})
+
+            DE/rand/2 : v_i = x_{r_1} + F_{xc}(x_{r_2} − x_{r_3})
```
Comment (andyfaff) [inline scipy/optimize/_differentialevolution.py:304]: On L260 we refer to population members as $b_0$, i.e. generalising to $b_x$. Can you change `x_{r_2}` to `b_{r_2}` in all these equations?

**Comment by mkiffer** [comment]
@andyfaff thanks for feedback will update these things now.

File: scipy/optimize/_differentialevolution.py (hunk 4 above)
Comment (mkiffer) [reply scipy/optimize/_differentialevolution.py:298]: done

Comment (mkiffer) [reply scipy/optimize/_differentialevolution.py:298]: done

File: scipy/optimize/_differentialevolution.py (hunk 6 above)
Comment (mkiffer) [reply scipy/optimize/_differentialevolution.py:304]: happy to change this, however we run into problems with the vector x_b which is supposed to represent the "best vector" and in the line260 notation we refer to the best vector as b_0. Do you think we should refer to best vector as b_b?

Alternatively we could change the notation on line 260 to x_b and I can change x_{r_2} --> population[rand1] for the remaining documentation to match the notation on line 260.

File: scipy/optimize/_differentialevolution.py (hunk 3 above)
Comment (mkiffer) [reply scipy/optimize/_differentialevolution.py:312]: I have edited all instances from `DE/current-to-best/1` to `currenttobest1*`

File: scipy/optimize/_differentialevolution.py (hunk 6 above)
Comment (andyfaff) [reply scipy/optimize/_differentialevolution.py:304]: The best vector is always kept in the first element of the population, b[0], $b_0$. Thus this equation should be rewritten:
```
b' = b_{r_1} + mutation * (b_{r_2} − b_{r_3}) 
```

Comment (mkiffer) [reply scipy/optimize/_differentialevolution.py:304]: I understand. Great, I will fix.

File: scipy/optimize/_differentialevolution.py (hunk 1 above)
Comment (mkiffer) [reply scipy/optimize/_differentialevolution.py]: not totally sure of the definition of NP. 

Is it popsize*N as described in L80 or is it just the size of our target vector b'?

**Comment by andyfaff** [comment]
The docs seem to render ok. I removed extra text from @mkiffer original commit because it was identical to text from a journal article, which would be under copyright.

**Pull request merged**